x/subscription/keeper/subscription_test.go

```diff
@@ -317,6 +320,49 @@ func TestSubscriptionAdminProject(t *testing.T) {
 	require.Nil(t, err)
 }
 
+func TestMonthlyRechargeCU(t *testing.T) {
+	ts := setupTestStruct(t, 1)
+	keeper := ts.keepers.Subscription
+	projectKeeper := ts.keepers.Projects
+
+	account := common.CreateNewAccount(ts._ctx, *ts.keepers, 10000)
+	creator := account.Addr.String()
+
+	err := keeper.CreateSubscription(ts.ctx, creator, creator, ts.plans[0].Index, 2, "")
+	require.Nil(t, err)
+
+	block := uint64(ts.ctx.BlockHeight())
+
+	sub, found := keeper.GetSubscription(ts.ctx, creator)
+	require.True(t, found)
+
+	// use the subscription and the project
+	keeper.ChargeSubscription(ts.ctx, creator, 1000)
+	require.Equal(t, sub.PrevCuLeft, sub.MonthCuTotal-1000)
+	err = projectKeeper.AddComputeUnitsToProject(ts.ctx, creator, block, 1000)
+	require.Nil(t, err)
+	// verify that project used the CU
+	proj, _, err := projectKeeper.GetProjectForDeveloper(ts.ctx, creator, block)
+	require.Nil(t, err)
+	require.Equal(t, uint64(1000), proj.UsedCu)
+
+	// fast-forward one months
+	sub = ts.expireSubscription(sub)
```

Comment:
also lets add a fixation in the middle of the month for the project and check that the old ones are ok

Reply:
done